federatedrec/svdpp/hetero_svdpp/hetero_svdpp_client.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#
+#  Copyright 2019 The FATE Authors. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import functools
+import typing
+
+import numpy as np
+from arch.api import session as fate_session
+from arch.api.utils import log_utils
+from federatedml.util import consts
+from federatedrec.optim.sync import user_ids_transfer_sync, average_rate_transfer_sync
+from federatedrec.svdpp.hetero_svdpp.hetero_svdpp_base import HeteroSVDppBase
+from federatedrec.svdpp.hetero_svdpp.backend import KerasSeqDataConverter, SVDppModel
+# from federatedrec.svdpp.hetero_svdpp.backend import TFSeqDataConverter, SVDppModel
+
+LOGGER = log_utils.getLogger()
+
+
+class HeteroSVDppClient(HeteroSVDppBase):
+    def __init__(self):
+        super(HeteroSVDppClient, self).__init__()
+
+        self._model = None
+
+    def _init_model(self, param):
+        super()._init_model(param)
+
+        self.batch_size = param.batch_size
+        self.aggregate_every_n_epoch = 1
+        self.optimizer = param.optimizer
+        self.loss = param.loss
```

Comment:
can parameter "loss" be set before training or be saved after training?

Reply:
loss if type of String, used to build model. Keras model use this parameter to choose the corresponding loss function, which can be saved after training.